### PR TITLE
[DON'T MERGE!!] Add ShadowRoot interface to BCD

### PR DESCRIPTION
**Pull request by chrisdavidmills**
See https://developer.mozilla.org/en-US/docs/Web/API/ShadowRoot

### CODE REVIEW

File: api/ShadowRoot.json [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+          "edge": {
+            "version_added": false,
+            "notes": "Under consideration"
+          },
+          "edge_mobile": {
+            "version_added": false,
+            "notes": "Under consideration"
+          },
+          "firefox": {
+            "version_added": true,
```
Comment (teoli2003) [inline api/ShadowRoot.json]: This is fairly recent, it should be easy to find the Fx version it landed.

File: api/ShadowRoot.json [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+              },
+              {
+                "type": "preference",
+                "name": "dom.webcomponents.shadowdom.enabled",
+                "value_to_set": "true"
+              }
+            ]
+          },
+          "firefox_android": {
+            "version_added": true,
```
Comment (teoli2003) [inline api/ShadowRoot.json]: This is fairly recent, it should be easy to find the Fx version it landed.

File: api/ShadowRoot.json [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+            "edge": {
+              "version_added": false,
+              "notes": "Under consideration"
+            },
+            "edge_mobile": {
+              "version_added": false,
+              "notes": "Under consideration"
+            },
+            "firefox": {
+              "version_added": true,
```
Comment (teoli2003) [inline api/ShadowRoot.json]: This is fairly recent, it should be easy to find the Fx version it landed.

File: api/ShadowRoot.json [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+                },
+                {
+                  "type": "preference",
+                  "name": "dom.webcomponents.shadowdom.enabled",
+                  "value_to_set": "true"
+                }
+              ]
+            },
+            "firefox_android": {
+              "version_added": true,
```
Comment (teoli2003) [inline api/ShadowRoot.json]: This is fairly recent, it should be easy to find the Fx version it landed.

File: api/ShadowRoot.json [hunk 5]
```diff
@@ -0,0 +1,355 @@
...
+            "edge": {
+              "version_added": false,
+              "notes": "Under consideration"
+            },
+            "edge_mobile": {
+              "version_added": false,
+              "notes": "Under consideration"
+            },
+            "firefox": {
+              "version_added": true,
```
Comment (teoli2003) [inline api/ShadowRoot.json]: This is fairly recent, it should be easy to find the Fx version it landed.

File: api/ShadowRoot.json [hunk 6]
```diff
@@ -0,0 +1,355 @@
...
+                },
+                {
+                  "type": "preference",
+                  "name": "dom.webcomponents.shadowdom.enabled",
+                  "value_to_set": "true"
+                }
+              ]
+            },
+            "firefox_android": {
+              "version_added": true,
```
Comment (teoli2003) [inline api/ShadowRoot.json]: This is fairly recent, it should be easy to find the Fx version it landed.

File: api/ShadowRoot.json [hunk 7]
```diff
@@ -0,0 +1,355 @@
...
+            "edge": {
+              "version_added": false,
+              "notes": "Under consideration"
+            },
+            "edge_mobile": {
+              "version_added": false,
+              "notes": "Under consideration"
+            },
+            "firefox": {
+              "version_added": true,
```
Comment (teoli2003) [inline api/ShadowRoot.json]: This is fairly recent, it should be easy to find the Fx version it landed.

File: api/ShadowRoot.json [hunk 8]
```diff
@@ -0,0 +1,355 @@
...
+                },
+                {
+                  "type": "preference",
+                  "name": "dom.webcomponents.shadowdom.enabled",
+                  "value_to_set": "true"
+                }
+              ]
+            },
+            "firefox_android": {
+              "version_added": true,
```
Comment (teoli2003) [inline api/ShadowRoot.json]: This is fairly recent, it should be easy to find the Fx version it landed.

File: api/ShadowRoot.json [hunk 9]
```diff
@@ -0,0 +1,355 @@
...
+            "edge": {
+              "version_added": true,
+              "notes": "Features still implemented on the <a href='https://developer.mozilla.org/docs/Web/API/Document'>Document</a> interface"
+            },
+            "edge_mobile": {
+              "version_added": true,
+              "notes": "Features still implemented on the <a href='https://developer.mozilla.org/docs/Web/API/Document'>Document</a> interface"
+            },
+            "firefox": {
+              "version_added": true,
```
Comment (teoli2003) [inline api/ShadowRoot.json]: If I read the note correctly, Document.documentorshadowroot exists but ShadowRoot.documentorshadowroot not. So this should be false, not true. (The comment is useful).

File: api/ShadowRoot.json [hunk 10]
```diff
@@ -0,0 +1,355 @@
...
+            },
+            "chrome": {
+              "version_added": "53"
+            },
+            "chrome_android": {
+              "version_added": "53"
+            },
+            "edge": {
+              "version_added": true,
+              "notes": "Features still implemented on the <a href='https://developer.mozilla.org/docs/Web/API/Document'>Document</a> interface"
```
Comment (teoli2003) [inline api/ShadowRoot.json]: If I read the note correctly, Document.documentorshadowroot exists but ShadowRoot.documentorshadowroot not. So this should be false, not true. (The comment is useful).

File: api/ShadowRoot.json [hunk 11]
```diff
@@ -0,0 +1,355 @@
...
+            },
+            "edge_mobile": {
+              "version_added": true,
+              "notes": "Features still implemented on the <a href='https://developer.mozilla.org/docs/Web/API/Document'>Document</a> interface"
+            },
+            "firefox": {
```
Comment (teoli2003) [inline api/ShadowRoot.json]: If I read the note correctly, Document.documentorshadowroot exists but ShadowRoot.documentorshadowroot not. So this should be false, not true. (The comment is useful).

File: api/ShadowRoot.json [hunk 12]
```diff
@@ -0,0 +1,355 @@
...
+              "version_added": true,
+              "notes": "Features still implemented on the <a href='https://developer.mozilla.org/docs/Web/API/Document'>Document</a> interface"
+            },
+            "firefox_android": {
+              "version_added": true,
```
Comment (teoli2003) [inline api/ShadowRoot.json]: If I read the note correctly, Document.documentorshadowroot exists but ShadowRoot.documentorshadowroot not. So this should be false, not true. (The comment is useful).

File: api/ShadowRoot.json [hunk 13]
```diff
@@ -0,0 +1,355 @@
...
+              "version_added": false
+            },
+            "opera": {
+              "version_added": "40"
+            },
+            "opera_android": {
+              "version_added": "40"
+            },
+            "safari": {
+              "version_added": true,
```
Comment (teoli2003) [inline api/ShadowRoot.json]: If I read the note correctly, Document.documentorshadowroot exists but ShadowRoot.documentorshadowroot not. So this should be false, not true. (The comment is useful).

File: api/ShadowRoot.json [hunk 14]
```diff
@@ -0,0 +1,355 @@
...
+              "notes": "Features still implemented on the <a href='https://developer.mozilla.org/docs/Web/API/Document'>Document</a> interface"
+            },
+            "safari_ios": {
+              "version_added": true,
```
Comment (teoli2003) [inline api/ShadowRoot.json]: If I read the note correctly, Document.documentorshadowroot exists but ShadowRoot.documentorshadowroot not. So this should be false, not true. (The comment is useful).

**Comment by chrisdavidmills** [comment]
I've done this work in https://github.com/mdn/browser-compat-data/pull/1075 — see over there for explanation of why.

**Comment by Elchi3** [comment]
Closing here then

**Pull request closed without merging**